explorer-nextjs/src/components/toggleButton/ToggleButton.tsx

```diff
@@ -12,17 +18,64 @@ type Options = [Option, Option];
 const ExplorerButtonGroup = ({
   size = "small",
   options,
+  onPage,
 }: {
   size?: "small" | "medium" | "large";
   options: Options;
+  onPage: string;
 }) => {
+  const [hasEpochStarted, setHasEpochStarted] = useState(false);
+  const [loading, setLoading] = useState<string | null>(null);
+
+  const { data } = useQuery({
+    enabled: true,
+    queryKey: ["currentEpoch"],
+    queryFn: fetchCurrentEpoch,
+    refetchInterval: 30000,
+    staleTime: 30000,
+    refetchOnMount: true,
+    keepPreviousData: false,
+  });
+
+  const queryClient = useQueryClient();
+  // check epoch update
+  useEffect(() => {
+    const checkEpochStatus = () => {
+      if (!data?.dateTime) return;
+      const oneHourLater = subSeconds(new Date(data.dateTime), 30).getTime();
+      const now = Date.now();
+      setHasEpochStarted(now >= oneHourLater);
+    };
+
+    checkEpochStatus();
+    const interval = setInterval(checkEpochStatus, 60000);
+    return () => clearInterval(interval);
+  });
```

Comment:
can you explain this as I'm not fully clear on what is going on or why we need this?

Reply:
why don't we check the current time against most recently fetched epoch time, if the currenttime > epochtime show "getting new epoch time" (or something like that). If we check the epoch every 30 or 60 secs eventually it's going to show correctly?

The frontend shouldnt resolving issues in epoch timing. If there's an issue with caching on the BE it should be fixed there.